src/Core/Resolvers/QueryExecutor.cs

```diff
@@ -200,7 +210,7 @@ public virtual void PopulateDbTypeForParameter(KeyValuePair<string, DbConnection
         }
 
         /// <inheritdoc />
-        public virtual async Task SetManagedIdentityAccessTokenIfAnyAsync(DbConnection conn)
+        public virtual async Task SetManagedIdentityAccessTokenIfAnyAsync(DbConnection conn, string? datasourceName = null)
```

Comment:
Can we use one of `dbName`/`dataSourceName` everywhere? Prompts me to think if they mean different things since we are in initial stages of this change.